tests/unit/test_hnsw_tiered.cpp

```diff
@@ -1899,9 +1829,288 @@ TYPED_TEST(HNSWTieredIndexTest, alternateInsertDeleteAsync) {
         // Verify that we have no pending jobs.
         EXPECT_EQ(tiered_index->labelToInsertJobs.size(), 0);
         EXPECT_EQ(tiered_index->idToRepairJobs.size(), 0);
-        EXPECT_EQ(tiered_index->idToSwapJob.size(), tiered_index->swapJobs.size());
-        for (auto job : tiered_index->swapJobs) {
-            EXPECT_EQ(job->pending_repair_jobs_counter.load(), 0);
+        EXPECT_EQ(tiered_index->idToSwapJob.size(), tiered_index->idToSwapJob.size());
+        for (auto &it : tiered_index->idToSwapJob) {
+            EXPECT_EQ(it.second->pending_repair_jobs_counter.load(), 0);
+        }
+        // Delete another vector to trigger swapping of vectors that hadn't been swapped yet.
+        // If the number of swap jobs is lower than the threshold, none of them are going to be
+        // executed, but otherwise, ALL of them should be executed.
+        size_t pending_swap_jobs = tiered_index->idToSwapJob.size();
+        EXPECT_EQ(tiered_index->deleteVector(0), 0);
```

Comment:
To trigger the swap jobs executions that are now ready...

Reply:
So in practice, if all deletion calls end while there are still pending repair jobs, and there are no further deletions, there might be vectors that will never be completely deleted (also happens if we never reach max swap jobs)
Maybe we should add an API function to enforce (if possible) swapping regardless of how many swap jobs are waiting. (The user can check that his job queue is empty and then call this function) 
Some kind of a garbage collector

Reply:
That's a good point. I guess that we could add the call for executing swap jobs from add vector as well so that if we accumulated many swap jobs, we would eventually execute them before we add new vectors and increase index capacity even more.

Reply:
I think that this addition would fit better in the next PR that enables support of overwrite, where we are checking if a label exists in HNSW in `addVector`

Reply:
This is also related to this comment:
https://github.com/RedisAI/VectorSimilarity/pull/349#discussion_r1165081312
Why is the responsibility of the deletion function that didn't cause any changes to the swap jobs queue, or insertion of a vector that happens to be at the right time, to trigger execution? It looks kind of arbitrary to me
We give the user the flexibility to determine this parameter of pending swap jobs but he can't really control it
Maybe it is better to just initialize it to the default value ❓

Reply:
It is arbitrary in a sense, and in practice, we don't expect users to be aware (or care) exactly when the swap job took place. It's our call. Also, the pending swap jobs threshold is configurable for our internal usage, we don't expose it to the user (for now)